### PR TITLE
pkg/middleware: more robust sessionid handling

### PR DESCRIPTION
**Pull request by sbinet**
Dots '.' may appear in the decoded sessionid string. The logic to extract the username from the decoded sessionid string would then be defeated.

### CODE REVIEW

**Comment by sbinet** [comment]
Perhaps a better way of handling this (trying to find where the username starts and where the key ends, is prone to errors: what if the username contains a `.` ?), would be to pass the expected username to the `Verifier` ?

File: pkg/middleware/auth.go [hunk 1]
```diff
@@ -25,22 +25,29 @@ func Verify(key string, noAuth bool) func(http.Handler) http.Handler {
...
-			decodedStr, err := b64.StdEncoding.DecodeString(ck.Value)
+			session, err := b64.StdEncoding.DecodeString(ck.Value)
 			if err != nil {
 				w.WriteHeader(400)
 				log.Println(err)
 				return
 			}
 
-			parsedStr := strings.Split(string(decodedStr), ".")
+			i := bytes.LastIndexByte(session, '.')
```
Comment (oxtyped) [inline pkg/middleware/auth.go:35]: Thanks for this!

**Review by oxtyped — APPROVED** [review]
lgtm!

**Comment by oxtyped** [comment]
> Perhaps a better way of handling this (trying to find where the username starts and where the key ends, is prone to errors: what if the username contains a . ?), would be to pass the expected username to the Verifier ?

I think this might be up for a separate discussion (maybe in a new issue). I'm not sure if we should support `.` in usernames since that might also have a bit of an effect when it comes to querying for usernames with `.` in the database iirc.

But should we choose either way, `pkg/apis/handlers.go` (https://github.com/oxtyped/gpodder2go/blob/main/pkg/apis/handlers.go#L50-L82) will still need to be updated to handle both scenarios.

**Pull request merged**